rust-mode.el

```diff
@@ -573,6 +584,54 @@ the desired identifiers), but does not match type annotations \"foo::<\"."
             ((not (looking-at (rx (0+ space) "<")))
 	     (throw 'rust-path-font-lock-matcher match))))))))
 
+(defun rust-next-string-interpolation (limit)
+  "Search forward from point for next Rust interpolation marker
+before LIMIT.
+Set point to the end of the occurrence found, and return match beginning
+and end."
+  (catch 'match
+    (save-match-data
+      (save-excursion
+        (while (search-forward "{" limit t)
+          (if (eql (char-after (point)) ?{)
```

Comment:
Ok, I see what you mean there, sorry I didn't understand that before.  It seems like it could still be done using a regexp with alternation, plus an explicit check for `{{` in the loop.

But ... mainly I wanted this to avoid a possibly-invalid call to `set-match-data`; but I looked through the Emacs sources and I found a couple other cases like this, so let's just go ahead.